### ISSUE TITLE
Error with “ns-install-cli” when creating fruitNeRF environment

### ISSUE THREAD

**Issue by mitsuki05** [meyerls/FruitNeRF#33]
The NeRFstudio environment was created with the following command.

`pip install torch==2.1.2+cu118 torchvision==0.16.2+cu118 --extra-index-url https://download.pytorch.org/whl/cu118`
`conda install -c "nvidia/label/cuda-11.8.0" cuda-toolkit`
`pip install ninja git+https://github.com/NVlabs/tiny-cuda-nn/#subdirectory=bindings/torch`
`pip install nerfstudio==0.3.2`
`pip install viser==0.1.24`

This properly created a NeRFstudio environment with version 0.3.2, and ns-train -h was also properly used.
(Using anaconda)

Then git clone fruitNeRF as per README, python -m pip install -e . Then I did ns-install-cli after
Then I got the following error in ns-install-cli.

```
Traceback (most recent call last):
  File "/home/krlabmember/anaconda3/envs/fruitNeRF/bin/ns-install-cli", line 8, in <module>
    sys.exit(entrypoint())
  File "/home/krlabmember/anaconda3/envs/fruitNeRF/lib/python3.8/site-packages/nerfstudio/scripts/completions/install.py", line 412, in entrypoint
    tyro.cli(main, description=__doc__)
  File "/home/krlabmember/anaconda3/envs/fruitNeRF/lib/python3.8/site-packages/tyro/_cli.py", line 229, in cli
    return run_with_args_from_cli()
  File "/home/krlabmember/anaconda3/envs/fruitNeRF/lib/python3.8/site-packages/nerfstudio/scripts/completions/install.py", line 393, in main
    _generate_completions_files(completions_dir, scripts_dir, shells_supported, shells_found)
  File "/home/krlabmember/anaconda3/envs/fruitNeRF/lib/python3.8/site-packages/nerfstudio/scripts/completions/install.py", line 324, in _generate_completions_files
    completion_paths = list(
  File "/home/krlabmember/anaconda3/envs/fruitNeRF/lib/python3.8/concurrent/futures/_base.py", line 619, in result_iterator
    yield fs.pop().result()
  File "/home/krlabmember/anaconda3/envs/fruitNeRF/lib/python3.8/concurrent/futures/_base.py", line 437, in result
    return self.__get_result()
  File "/home/krlabmember/anaconda3/envs/fruitNeRF/lib/python3.8/concurrent/futures/_base.py", line 389, in __get_result
    raise self._exception
  File "/home/krlabmember/anaconda3/envs/fruitNeRF/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/krlabmember/anaconda3/envs/fruitNeRF/lib/python3.8/site-packages/nerfstudio/scripts/completions/install.py", line 326, in <lambda>
    lambda path_or_entrypoint_and_shell: _generate_completion(
  File "/home/krlabmember/anaconda3/envs/fruitNeRF/lib/python3.8/site-packages/nerfstudio/scripts/completions/install.py", line 129, in _generate_completion
    raise e
  File "/home/krlabmember/anaconda3/envs/fruitNeRF/lib/python3.8/site-packages/nerfstudio/scripts/completions/install.py", line 116, in _generate_completion
    new = subprocess.run(
  File "/home/krlabmember/anaconda3/envs/fruitNeRF/lib/python3.8/subprocess.py", line 516, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['ns-train', '--tyro-print-completion', 'zsh']' returned non-zero exit status 1.
``` 

I also got the following error with ns-train -h

``` 
        geo_feat_dim: 30 has type <class 'fruit_nerf.fruit_pipeline.FruitPipelineConfig'>. We'll try to handle this gracefully, but it may cause unexpected behavior.
  warnings.warn(
/home/krlabmember/anaconda3/envs/fruitNeRF/lib/python3.8/site-packages/tyro/_fields.py:1076: UserWarning: Mutable type <class 'nerfstudio.data.datamanagers.base_datamanager.DataManagerConfig'> is used as a default value for `datamanager`. This is dangerous! Consider using `dataclasses.field(default_factory=...)` or marking <class 'nerfstudio.data.datamanagers.base_datamanager.DataManagerConfig'> as frozen.
  warnings.warn(
/home/krlabmember/anaconda3/envs/fruitNeRF/lib/python3.8/site-packages/tyro/_fields.py:1076: UserWarning: Mutable type <class 'nerfstudio.models.base_model.ModelConfig'> is used as a default value for `model`. This is dangerous! Consider using `dataclasses.field(default_factory=...)` or marking <class 'nerfstudio.models.base_model.ModelConfig'> as frozen.
  warnings.warn(
Traceback (most recent call last):
  File "/home/krlabmember/anaconda3/envs/fruitNeRF/bin/ns-train", line 8, in <module>
    sys.exit(entrypoint())
  File "/home/krlabmember/anaconda3/envs/fruitNeRF/lib/python3.8/site-packages/nerfstudio/scripts/train.py", line 262, in entrypoint
    tyro.cli(
  File "/home/krlabmember/anaconda3/envs/fruitNeRF/lib/python3.8/site-packages/tyro/_cli.py", line 207, in cli
    output = _cli_impl(
  File "/home/krlabmember/anaconda3/envs/fruitNeRF/lib/python3.8/site-packages/tyro/_cli.py", line 408, in _cli_impl
    parser_spec = _parsers.ParserSpecification.from_callable_or_type(
  File "/home/krlabmember/anaconda3/envs/fruitNeRF/lib/python3.8/site-packages/tyro/_parsers.py", line 114, in from_callable_or_type
    field_out = handle_field(
  File "/home/krlabmember/anaconda3/envs/fruitNeRF/lib/python3.8/site-packages/tyro/_parsers.py", line 320, in handle_field
    subparsers_attempt = SubparsersSpecification.from_field(
  File "/home/krlabmember/anaconda3/envs/fruitNeRF/lib/python3.8/site-packages/tyro/_parsers.py", line 526, in from_field
    subparser = ParserSpecification.from_callable_or_type(
  File "/home/krlabmember/anaconda3/envs/fruitNeRF/lib/python3.8/site-packages/tyro/_parsers.py", line 114, in from_callable_or_type
    field_out = handle_field(
  File "/home/krlabmember/anaconda3/envs/fruitNeRF/lib/python3.8/site-packages/tyro/_parsers.py", line 320, in handle_field
    subparsers_attempt = SubparsersSpecification.from_field(
  File "/home/krlabmember/anaconda3/envs/fruitNeRF/lib/python3.8/site-packages/tyro/_parsers.py", line 469, in from_field
    assert default_name is not None, (
AssertionError: `pipeline` was provided a default value of type <class 'fruit_nerf.fruit_pipeline.FruitPipelineConfig'> but no matching subcommand was found. A type may be missing in the Union type declaration for `pipeline`, which currently expects [<class 'nerfstudio.pipelines.base_pipeline.VanillaPipelineConfig'>, <class 'fruit_nerf.fruit_pipeline.FruitPipelineConfig'>]. The types may also be too complex for tyro's subcommand matcher; support is particularly limited for custom generic types.
``` 

Do you have any idea what these errors are?

**Comment by sebdeveloper6952** [comment]
@mitsuki05  i had the same error and was able to move past it, referencing this other issue: https://github.com/nerfstudio-project/nerfstudio/issues/3491

TL;DR
run `pip install tyro==0.8.12`

**Comment by mitsuki05** [comment]
I was able to create an environment well with this method!! thank you
